src/Server/TCPHandler.cpp

```diff
@@ -630,6 +627,8 @@ void TCPHandler::runImpl()
             state.io.onException();
             exception.reset(e.clone());
 
+            state.timeout_setter.reset();
```

Comment:
> Though state.reset() will not be reset in case of is_interserver_mode because we rethrow the exception in this case before resetting the state... So I think better to also reset the timeout right after the last catch of this try block

we rethrow in case of `is_interserver_mode` AND `AUTHENTICATION_FAILED` - which means we don't need to reset after last catch, I guess?

Reply:
Yes indeed